web/react/utils/utils.jsx

```diff
@@ -437,60 +528,86 @@ module.exports.textToJsx = function(text, options) {
                 // do both a non-case sensitive and case senstive check
                 var mClass = implicitKeywords.indexOf('@'+name.toLowerCase()) !== -1 || implicitKeywords.indexOf('@'+name) !== -1 ? mentionClass : "";
 
-                var suffix = word.match(puncEndRegex);
-                var prefix = word.match(puncStartRegex);
-
                 if (searchTerm === name) {
                     highlightSearchClass = " search-highlight";
                 }
 
-                inner.push(<span key={name+i+z+"_span"}>{prefix}<a className={mClass + highlightSearchClass + " mention-link"} key={name+i+z+"_link"} href="#" onClick={function(value) { return function() { module.exports.searchForTerm(value); } }(name)}>@{name}</a>{suffix} </span>);
+                if (useMarkdown) {
+                    prefixSpan ? inner.push(prefixSpan) : null;
+                    inner.push(<span key={word+i+z+"word_span"}><a className={mClass + highlightSearchClass + " mention-link"} key={name+i+z+"_link"} href="#" dangerouslySetInnerHTML={{__html: "@" + name}} onClick={function(value) { return function() { module.exports.searchForTerm(value); } }(name)} /></span>);
+                    suffixSpan ? inner.push(suffixSpan) : null;
+                }
+                else
```

Comment:
Not sure when we'll be fully enforcing the style guide, but this else and some of the later ones should have the curly braces fixed so that it matches when we do switch it on.